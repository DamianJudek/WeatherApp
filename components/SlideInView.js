import React from 'react';
import { Animated } from 'react-native';
import PropTypes from 'prop-types';

class SlideInView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      opacity: new Animated.Value(0),
    };
  }

  static propTypes = {
    style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
    children: PropTypes.array,
    delay: PropTypes.number,
  };

  fadeIn = () => {
    const { delay } = this.props;
    Animated.timing(this.state.opacity, {
      toValue: 1,
      duration: 750,
      useNativeDriver: true,
      delay: delay || 0,
    }).start();
  };

  componentDidMount() {
    this.fadeIn();
  }

  render() {
    return (
      <Animated.View
        style={[
          this.props.style,
          {
            opacity: this.state.opacity,
            transform: [
              {
                translateY: this.state.opacity.interpolate({
                  inputRange: [0, 1],
                  outputRange: [35, 0],
                }),
              },
            ],
          },
        ]}>
        {this.props.children}
      </Animated.View>
    );
  }
}

export default SlideInView;
