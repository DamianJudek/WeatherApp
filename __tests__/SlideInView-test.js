import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SlideInView from '../components/SlideInView';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
test('renders correctly', () => {
  const snap = renderer.create(<SlideInView />).toJSON();
  expect(snap).toMatchSnapshot();
});
