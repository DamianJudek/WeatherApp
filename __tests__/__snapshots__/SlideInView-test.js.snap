// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "opacity": 0,
      "transform": Array [
        Object {
          "translateY": 35,
        },
      ],
    }
  }
/>
`;
